test/parallel/test-vm-cached-data.js

```diff
@@ -2,19 +2,36 @@
 require('../common');
 const assert = require('assert');
 const vm = require('vm');
+const spawnSync = require('child_process').spawnSync;
 const Buffer = require('buffer').Buffer;
 
 function getSource(tag) {
   return `(function ${tag}() { return \'${tag}\'; })`;
 }
 
-function produce(source) {
-  const script = new vm.Script(source, {
-    produceCachedData: true
-  });
-  assert(!script.cachedDataProduced || script.cachedData instanceof Buffer);
+function produce(source, count) {
+  if (!count)
+    count = 1;
+
+  const out = spawnSync(process.execPath, [ '-p', `
+    var assert = require('assert');
+    var vm = require('vm');
+
+    for (var i = 0; i < ${count}; i++) {
+      var script = new vm.Script(process.argv[1], {
+        produceCachedData: true
+      });
+
+      assert(!script.cachedDataProduced || script.cachedData instanceof Buffer);
+
+      if (script.cachedDataProduced)
+        script.cachedData.toString('base64');
```

Comment:
Hm, very subtle.  I guess it's not wrong but it's not very obviously correct, either.  I'd use `-e` and explicitly print it.